code/analyticsedge/src/phone/java/com/adobe/marketing/mobile/AnalyticsExtension.java

```diff
@@ -202,6 +225,65 @@ public void handleAnalyticsTrackEvent(final Event event) {
         }
     }
 
+    /**
+     * Adds the Rules Engine Response Content event to the event queue to be processed.
+     *
+     * <p>
+     * Any events in the event queue will be cleared if the privacy status is opted out.
+     *
+     * @param event The Rules Engine Response Content {@link Event} to be processed.
+     */
+    @Override
+    public void handleRulesEngineEvent(final Event event) {
+        if (event == null) {
+            Log.debug(AnalyticsConstants.LOG_TAG, "Unable to handle rules engine response. Event received is null.");
+            return;
+        }
+
+        eventQueue.add(event);
+        processEvents();
+    }
+
+    /**
+     * Processes the passed in Rules Engine Response Content event.
+     *
+     * @param event The Rules Engine Response Content {@link Event} to be processed.
+     */
+    private void processRulesEngineResponseEvent(final Event event) {
+        Map<String, Object> data = event.getEventData();
+        if (data == null || data.isEmpty()) {
+            Log.debug(LOG_TAG, "processRulesEngineResponseEvent - Event with id %s contained no data, ignoring.", event.getUniqueIdentifier());
+            return;
+        }
+
+        Log.trace(LOG_TAG, "processRulesEngineResponseEvent - Processing event with id %s.", event.getUniqueIdentifier());
+
+        final Map<String,Object> consequence = (Map<String,Object>) data.get(AnalyticsConstants.EventDataKeys.TRIGGERED_CONSEQUENCE);
+        if (consequence == null || consequence.isEmpty()) {
+            Log.trace(LOG_TAG, "processRulesEngineResponseEvent - Ignoring as missing consequence data in event with id %s.", event.getUniqueIdentifier());
+            return;
+        }
+
+        final String consequenceType = (String) consequence.get(AnalyticsConstants.EventDataKeys.TYPE);
+        if (!AnalyticsConstants.ConsequenceTypes.TRACK.equals(consequenceType)) {
+            Log.trace(LOG_TAG, "processRulesEngineResponseEvent - Ignoring as consequence type is not analytics in event with id %s.", event.getUniqueIdentifier());
+            return;
+        }
+
+        final String consequenceId = (String) consequence.get(AnalyticsConstants.EventDataKeys.ID);
+        if (StringUtils.isNullOrEmpty(consequenceId)) {
+            Log.trace(LOG_TAG, "processRulesEngineResponseEvent - Ignoring as consequence id is missing in event with id %s.", event.getUniqueIdentifier());
+            return;
+        }
+
+        Map<String, Object> consequenceDetail = new HashMap<>();
```

Comment:
Can we initialize this map with Collections.EMPTY_MAP instead of creating a new instance.

Reply:
sure, i'l get this updated prior to next beta release.